include/ginkgo/core/multigrid/multigrid_level.hpp

```diff
@@ -52,51 +52,75 @@ namespace multigrid {
 /**
  * This class represents two levels in a multigrid hierarchy.
  *
- * A class implementing this interface should use the
- * this->get_compositions()->apply(...) as its own apply, which represents
- * op(b) = prolong(coarse(restrict(b))). All MultigridLevel generator should
- * inherit from this class.
- *
- * @ingroup Multigrid
+ * The MultigridLevel is an interface that allows to get the individual
+ * components of multigrid level. Each Linop for MultigridLevel should inherit
+ * from this interface. Use EnableMultigridLevel<ValueType> to implement this
+ * interface with composition by default.
  */
-template <typename ValueType>
-class MultigridLevel : public UseComposition<ValueType> {
+class MultigridLevel {
```

Comment:
Why is a `MultigridLevel` without `ValueType` needed? Did you realize this was necessary for another pull request?